src/main/react/app/routes/builder/context/node-context.tsx

```diff
@@ -0,0 +1,110 @@
+import useNodeContextStore from '~/stores/node-context-store'
+import { useEffect, useRef, useState } from 'react'
+import useFlowStore from '~/stores/flow-store'
+
+export default function NodeContext({
+  nodeId,
+  setShowNodeContext,
+}: Readonly<{
+  setShowNodeContext: (b: boolean) => void
+  nodeId: number
+}>) {
+  const attributes = useNodeContextStore((state) => state.attributes)
+  const inputReferences = useRef<Record<number, HTMLInputElement | null>>({})
+  const { setAttributes, setNodeName } = useFlowStore((state) => state)
```

Comment:
I don't think this arrow function in needed when selecting the whole state.
Maybe selecting stuff isn't a bad idea, to minimize component re-renders. Otherwise, we could probably do a refactor afterward. (Thinking of it, re-renders might not really be a problem in this component, but still a good practice)